src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -230,59 +290,64 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     // - Populates the provided nameMap with all of our actions and our parents actions
     //    while omitting the actions that were already added before
     // Arguments:
-    // - nameMap: the nameMap we're populating. This maps the name (hstring) of a command to the command itself.
-    //             There should only ever by one of each command (identified by the actionID) in the nameMap.
+    // - nameMap: the nameMap we're populating, this maps the name (hstring) of a command to the command itself
     void ActionMap::_PopulateNameMapWithStandardCommands(std::unordered_map<hstring, Model::Command>& nameMap) const
     {
-        std::unordered_set<InternalActionID> visitedActionIDs;
-        for (const auto& cmd : _GetCumulativeActions())
+        for (const auto& [_, cmd] : _CumulativeActionMapCache)
         {
-            // only populate with valid commands
-            if (cmd.ActionAndArgs().Action() != ShortcutAction::Invalid)
+            const auto& name{ cmd.Name() };
+            if (!name.empty())
             {
-                // Only populate NameMap with actions that haven't been visited already.
-                const auto actionID{ Hash(cmd.ActionAndArgs()) };
-                if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
+                // there might be a collision here, where there could be 2 different commands with the same name
+                // in this case, prioritize the user's action
+                // TODO GH #17166: we should no longer use Command.Name to identify commands anywhere
+                if (!nameMap.contains(name) || cmd.Origin() == OriginTag::User)
                 {
-                    const auto& name{ cmd.Name() };
-                    if (!name.empty())
-                    {
-                        // Update NameMap.
-                        nameMap.insert_or_assign(name, cmd);
-                    }
-
-                    // Record that we already handled adding this action to the NameMap.
-                    visitedActionIDs.emplace(actionID);
+                    // either a command with this name does not exist, or this is a user-defined command with a name
+                    // in either case, update the name map with the command (if this is a user-defined command with
+                    // the same name as an existing command, the existing one will get overwritten)
+                    nameMap.insert_or_assign(name, cmd);
                 }
             }
         }
     }
 
     // Method Description:
-    // - Provides an accumulated list of actions that are exposed. The accumulated list includes actions added in this layer, followed by actions added by our parents.
-    std::vector<Model::Command> ActionMap::_GetCumulativeActions() const noexcept
+    // - Recursively populate keyBindingsMap with ours and our parents' key -> id pairs
+    // - This is a bottom-up approach, ensuring that the keybindings of the parents are overridden by the children

```

Comment:
this approach is bottoms-up, but that is _not_ what ensures parents get overridden by children. The code on line R322 does that. :P
That's distinct from it being a bottoms-up approach